synapse/handlers/identity.py

```diff
@@ -55,6 +55,14 @@ def __init__(self, hs):
         self.federation_http_client = hs.get_federation_http_client()
         self.hs = hs
 
+        # Only allow the web client location to be sent if it is an HTTP URL.
+        self._web_client_location = None
+        webclient_loc = hs.config.web_client_location
+        if webclient_loc and (
+            webclient_loc.startswith("http://") or webclient_loc.startswith("https://")
+        ):
```

Comment:
Should we not check this in the config parsing?

Reply:
This config option is allowed to be a web location (https?://) *or* a location on disk, in which case we serve it via Synapse.

See the code around https://github.com/matrix-org/synapse/blob/be2db93b3c14396d53d30f8d5f92db014453487b/synapse/app/homeserver.py#L261-L280

Reply:
(I also wouldn't be opposed to removing this ability, but that seems like a separate PR. 😄 )

Reply:
Oooooh, right yes.